### PR TITLE
Added middleware to publisher and subscriber

### PR DESCRIPTION
**Pull request by mhuggins**
@brandonc This implements the middleware layer for publisher and subscriber.  /cc @paul 

Side note: the configuration is getting pretty bloated.  Might want to reconsider that down the road.

### CODE REVIEW

File: lib/circuitry/publisher.rb [hunk 1]
```diff
@@ -30,10 +30,12 @@ def publish(topic_name, object)
       raise ArgumentError, 'object cannot be nil' if object.nil?
       raise PublishError, 'AWS configuration is not set' unless can_publish?
 
+      message = object.to_json
```
Comment (brandonc) [inline lib/circuitry/publisher.rb]: whoa is this broken previously

Comment (brandonc) [inline lib/circuitry/publisher.rb]: Oh i see now we were doing this in publish_internal before

**Comment by brandonc** [comment]
Really nice don't forget to bump version and add changelog

**Comment by mhuggins** [comment]
Yep yep, will do after merge.

**Pull request merged**